kernel/sched.c

```diff
@@ -657,24 +659,103 @@ void z_sched_start(struct k_thread *thread)
 	z_reschedule(&sched_spinlock, key);
 }
 
+static void z_thread_halt(struct k_thread *thread, k_spinlock_key_t key,
+			  bool terminate)
```

Comment:
Can you arrange this refactoring such that the renaming of z_thread_abort() to z_thread_halt() happens in place so it doesn't look like a big cut/paste patch?  Alternatively if that can't be arranged, can you split the patches to separate the motion of the code form the renaming?

Reply:
I'll try.